code-studio/src/js/progress.js

```diff
@@ -59,8 +61,25 @@ progress.populateProgress = function (scriptName) {
       if (serverProgress[levelId].submitted) {
         // Clear the existing class and replace
         $('.level-' + levelId).attr('class', 'level_link submitted');
+      } else if (serverProgress[levelId].pages_completed) {
+        // This is a multi-page level.  There will be multiple dots for the same level ID,
+        // so we need to decorate each of them individually.
+        var pagesCompleted = serverProgress[levelId].pages_completed;
```

Comment:
I'm not sure the right thing is going to happen if we are in HOC scale mode and aren't posting to the server on every level completion-- we would need to be able to record multi-page progress on the client and render from the client state.

Can we discuss live?